include/SDL3/SDL_gpu.h

```diff
@@ -0,0 +1,2251 @@
+
+/**
+ * Creates a pipeline object to be used in a graphics workflow.
+ *
+ * \param device a GPU Context
+ * \param pipelineCreateInfo a struct describing the state of the desired graphics pipeline
+ * \returns a graphics pipeline object on success, or NULL on failure
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuCreateShader
+ * \sa SDL_GpuBindGraphicsPipeline
+ * \sa SDL_GpuReleaseGraphicsPipeline
+ */
+extern SDL_DECLSPEC SDL_GpuGraphicsPipeline *SDLCALL SDL_GpuCreateGraphicsPipeline(
+    SDL_GpuDevice *device,
+    SDL_GpuGraphicsPipelineCreateInfo *pipelineCreateInfo);
+
+/**
+ * Creates a sampler object to be used when binding textures in a graphics workflow.
+ *
+ * \param device a GPU Context
+ * \param samplerCreateInfo a struct describing the state of the desired sampler
+ * \returns a sampler object on success, or NULL on failure
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuBindVertexSamplers
+ * \sa SDL_GpuBindFragmentSamplers
+ * \sa SDL_ReleaseSampler
+ */
+extern SDL_DECLSPEC SDL_GpuSampler *SDLCALL SDL_GpuCreateSampler(
+    SDL_GpuDevice *device,
+    SDL_GpuSamplerCreateInfo *samplerCreateInfo);
+
+/**
+ * Creates a shader to be used when creating a graphics pipeline.
+ *
+ * Shader resource bindings must be authored to follow a particular order.
+ * For SPIR-V shaders, use the following resource sets:
+ *  For vertex shaders:
+ *   0: Sampled textures, followed by storage textures, followed by storage buffers
+ *   1: Uniform buffers
+ *  For fragment shaders:
+ *   2: Sampled textures, followed by storage textures, followed by storage buffers
+ *   3: Uniform buffers
+ *
+ * For HLSL/DXBC/DXIL, use the following order:
+ *  For t registers:
+ *   Sampled textures, followed by storage textures, followed by storage buffers
+ *  For s registers:
+ *   Samplers with indices corresponding to the sampled textures
+ *  For b registers:
+ *   Uniform buffers
+ *
+ * For MSL/metallib, use the following order:
+ *  For [[texture]]:
+ *   Sampled textures, followed by storage textures
+ *  For [[sampler]]:
+ *   Samplers with indices corresponding to the sampled textures
+ *  For [[buffer]]:
+ *   Uniform buffers, followed by storage buffers.
+ *   Vertex buffer 0 is bound at [[buffer(30)]], vertex buffer 1 at [[buffer(29)]], and so on.
+ *    Rather than manually authoring vertex buffer indices, use the [[stage_in]] attribute
+ *    which will automatically use the vertex input information from the SDL_GpuPipeline.
+ *
+ * \param device a GPU Context
+ * \param shaderCreateInfo a struct describing the state of the desired shader
+ * \returns a shader object on success, or NULL on failure
+ *
+ * \since This function is available since SDL 3.x.x
+ *
+ * \sa SDL_GpuCreateGraphicsPipeline
+ * \sa SDL_GpuReleaseShader
+ */
+extern SDL_DECLSPEC SDL_GpuShader *SDLCALL SDL_GpuCreateShader(
+    SDL_GpuDevice *device,
+    SDL_GpuShaderCreateInfo *shaderCreateInfo);
```

Comment:
`SDL_GpuCreateShaderWithProperties()` for future expansion?

Reply:
Tracking at https://github.com/thatcosmonaut/SDL/issues/157